lib/FusionInventory/Agent/Task/Inventory/Win32/Firewall.pm

```diff
@@ -0,0 +1,201 @@
+            PROFILE => $subKeys->{$profile}
+        };
+    }
+
+    return $profiles;
+}
+
+sub _makeProfileAndConnectionsAssociation {
+    my (%params) = @_;
+
+    return unless $params{firewallProfiles};
+
+    my ($profilesKey, $signaturesKey) = _retrieveProfilesAndSignaturesKey();
+    return unless $profilesKey && $signaturesKey;
+
+    foreach my $interface (getInterfaces(
+        additionalPropertiesNetWorkAdapterConfiguration => [qw/DNSDomain/],
+        additionalPropertiesNetWorkAdapter => [qw/GUID/]
+    )) {
+        next if ($interface->{STATUS} ne 'Up');
+
+        my $profile;
+
+        my $domainSettings = _getConnectionDomainSettings(
+            guid => $interface->{GUID}
+        );
+        # check if connection with domain
+        if ($domainSettings) {
+            $profile = _retrieveFirewallProfileWithdomain(
+                profileName => $domainSettings->{'/PrimaryDomainName'},
+                profilesKey => $profilesKey
+            )
+        } else {
+            $profile = _retrieveFirewallProfileWithoutDomain(
+                DNSDomain => $interface->{DNSDomain},
+                profilesKey => $profilesKey,
+                signaturesKey => $signaturesKey
+            );
+        }
+
+        next unless $profile;
+	
+        my $category = hex2dec($profile->{'/Category'});
+        if (not defined $params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS}) {
+            $params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS} = [];
+        }
+        my $connection = {DESCRIPTION => $interface->{DESCRIPTION}};
+        $connection->{IPADDRESS} = $interface->{IPADDRESS} if ($interface->{IPADDRESS});
+        $connection->{IPADDRESS6} = $interface->{IPADDRESS6} if ($interface->{IPADDRESS6});
+        push @{$params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS}}, $connection;
+    }
+
+    my @profiles = ();
+    for my $p (values %{$params{firewallProfiles}}) {
+        my @p;
+        if ($p->{CONNECTIONS} && ref($p->{CONNECTIONS}) eq 'ARRAY') {
+            my @conns = @{$p->{CONNECTIONS}};
+            delete $p->{CONNECTIONS};
+            for my $conn (@conns) {
+                my $newP = dclone $p;
+                while (my ($k, $v) = each %$conn) {
+                    $newP->{$k} = $v;
+                }
+                push @p, $newP;
+            }
+        } else {
+            push @p, $p;
+        }
+        push @profiles, @p;
+    }
+
+    return @profiles;
+}
+
+sub _getConnectionDomainSettings {
+    my (%params) = @_;
+
+    return unless $params{guid};
+
+    my $registeredAdapter = getRegistryKey(
+        logger => $params{logger},
+        path => 'HKEY_LOCAL_MACHINE/SYSTEM/CurrentControlSet/services/Tcpip/Parameters/DNSRegisteredAdapters/' . $params{guid}
+    );
+    if ($registeredAdapter && $registeredAdapter->{'/PrimaryDomainName'}) {
+        return $registeredAdapter;
+    } else {
```

Comment:
I don't see the need of a "else { return; }" as last case at the end of a method, or clearly return a default value without the "else"